### PR TITLE
feat(provider/google): Support UHCs in L7 load balancers.

### PR DESCRIPTION
**Pull request by jtk54**

### CODE REVIEW

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/UpsertGoogleHttpLoadBalancerAtomicOperation.groovy [hunk 1]
```diff
@@ -119,9 +119,16 @@ class UpsertGoogleHttpLoadBalancerAtomicOperation extends UpsertGoogleLoadBalanc
     String urlMapName = httpLoadBalancer?.urlMapName ?: httpLoadBalancerName // An L7 load balancer is identified by its UrlMap name in Google Cloud Console.
 
     // Get all the existing infrastructure.
-    Set<HttpHealthCheck> existingHealthChecks = timeExecute(
-        compute.httpHealthChecks().list(project),
-        "compute.httpHealthChecks.list",
+
+    // Look up the legacy health checks so we can do the work to transition smoothly to the UHCs.
+    Set<HttpHealthCheck> legacyHealthChecks = timeExecute(
+      compute.httpHealthChecks().list(project),
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/UpsertGoogleHttpLoadBalancerAtomicOperation.groovy]: The indentation doesn't match between these 2 blocks.

File: clouddriver-google/src/test/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/UpsertGoogleHttpLoadBalancerAtomicOperationUnitSpec.groovy [hunk 2]
```diff
@@ -246,12 +254,15 @@ class UpsertGoogleHttpLoadBalancerAtomicOperationUnitSpec extends Specification
...
-          targetLink: "health-check",
-          name: HEALTH_CHECK_OP_NAME,
-          status: DONE)
+
+      def healthChecks = Mock(Compute.HealthChecks)
+      def healthChecksList = Mock(Compute.HealthChecks.List)
+      def healthCheckListReal = new HealthCheckList(items: [])
+      def healthChecksInsert = Mock(Compute.HealthChecks.Insert)
+      def healthChecksInsertOp = new Operation(
+      targetLink: "health-check",
```
Comment (duftler) [inline clouddriver-google/src/test/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/UpsertGoogleHttpLoadBalancerAtomicOperationUnitSpec.groovy]: Indentation is off here.

**Comment by duftler** [comment]
Ping.

**Comment by jtk54** [comment]
Waiting on @danielpeach to get around to the frontend changes, likely this week.

**Comment by anotherchrisberry** [comment]
ping

**Comment by jtk54** [comment]
Closing for now -- will reopen when we have time to land the frontend changes as well.

**Pull request closed without merging**